packages/swing-store/src/bundleStore.js

```diff
@@ -0,0 +1,242 @@
+   * Bundle artifact names should be strings of the form:
+   *   `bundle.${bundleID}`
+   *
+   * @param {string} name
+   *
+   * @yields {Uint8Array}
+   * @returns {AsyncIterable<Uint8Array>}
+   */
+  async function* exportBundle(name) {
+    typeof name === 'string' || Fail`artifact name must be a string`;
+    const parts = name.split('.');
+    const [type, bundleID] = parts;
+    // prettier-ignore
+    (parts.length === 2 && type === 'bundle') ||
+      Fail`expected artifact name of the form 'bundle.{bundleID}', saw ${q(name)}`;
+    const bundle = getBundle(bundleID);
+    bundle || Fail`bundle ${q(name)} not available`;
+    yield* Readable.from(Buffer.from(bundle.endoZipBase64));
+  }
+
+  const sqlGetBundleIDs = db.prepare(`
+    SELECT bundleID
+    FROM bundles
+  `);
+  sqlGetBundleIDs.pluck(true);
+
+  /**
+   * Obtain artifact metadata records for bundles contained in this store.
+   *
+   * @yields {[key: string, value: string]}
+   * @returns {Iterable<[key: string, value: string]>}
+   */
+  function* getExportRecords() {
+    for (const bundleID of sqlGetBundleIDs.iterate()) {
+      yield [bundleArtifactName(bundleID), bundleID];
+    }
+  }
+
+  async function* getArtifactNames() {
+    for (const bundleID of sqlGetBundleIDs.iterate()) {
+      yield bundleArtifactName(bundleID);
+    }
+  }
+
+  function computeSha512(bytes) {
+    const hash = createHash('sha512');
+    hash.update(bytes);
+    return hash.digest().toString('hex');
+  }
+
+  /**
+   * @param {string} name  Artifact name of the bundle
+   * @param {SwingStoreExporter} exporter  Whence to get the bits
+   * @param {string} bundleID  Bundle ID of the bundle
+   * @returns {Promise<void>}
+   */
+  async function importBundle(name, exporter, bundleID) {
+    const parts = name.split('.');
+    const [type, bundleIDkey] = parts;
+    // prettier-ignore
+    parts.length === 2 && type === 'bundle' ||
+      Fail`expected artifact name of the form 'bundle.{bundleID}', saw '${q(name)}'`;
+    bundleIDkey === bundleID ||
+      Fail`bundle artifact name ${name} doesn't match bundleID ${bundleID}`;
+    bundleID.startsWith('b1-') || Fail`invalid bundleID ${q(bundleID)}`;
+    const artifactChunks = exporter.getArtifact(name);
+    const inStream = Readable.from(artifactChunks);
+    const encodedBundle = await buffer(inStream);
+    const bundle = harden({
+      moduleFormat: 'endoZipBase64',
+      endoZipBase64Sha512: bundleID.substring(3),
+      endoZipBase64: encodedBundle.toString(),
+    });
+    await checkBundle(bundle, computeSha512, bundleID);
+    addBundle(bundleID, bundle);
+  }
+
+  const sqlDumpBundles = db.prepare(`
+    SELECT bundleID, bundle
+    FROM bundles
+    ORDER BY bundleID
+  `);
+
+  /**
+   * debug function to dump active bundles
+   */
+  function dumpBundles() {
+    const sql = sqlDumpBundles;
+    const dump = {};
+    for (const row of sql.iterate()) {
+      const { bundleID, bundle } = row;
+      dump[bundleID] = bundle;
```

Comment:
OTOH, this `dump()` format might want to be base64.. I know sometimes we do a quick `console.log(JSON.stringify(swingStore.dump()))`, although in practice if there are any non-trivial bundles in play, that'll be mostly useless to page through. But we might want to say that `dump()` is at least printable, and the raw DB rows are (appropriately) binary.